pinot-core/src/main/java/com/linkedin/pinot/core/plan/maker/InstancePlanMakerImplV2.java

```diff
@@ -141,13 +141,17 @@ private boolean isFitForMetadataBasedPlan(BrokerRequest brokerRequest) {
     }
 
     List<AggregationInfo> aggregationsInfo = brokerRequest.getAggregationsInfo();
-    if (aggregationsInfo != null && aggregationsInfo.size() == 1 && !brokerRequest.isSetGroupBy()) {
-      if (aggregationsInfo.get(0)
-          .getAggregationType()
-          .equalsIgnoreCase(AggregationFunctionFactory.AggregationFunctionType.COUNT.getName())) {
-        return true;
+    if (aggregationsInfo != null && !brokerRequest.isSetGroupBy()) {
```

Comment:
These are bunch of optimizers, we need to have a better framework, instead of calling isFitForStarTree, isFitForMetadata, isFitForDictionary. Not really part of this PR but please create an issue to track this later.